### PR TITLE
Fix various issues with Noetic build

### LINKED ISSUE #2322: IKfast plugin not working
### Description

Hi, I installed the new version of Moveit for Noetic, but can't make IKFast plugin work.
Running the package as it is I get this error:

```
Traceback (most recent call last):
  File "/home/usr/catkin_ws/src/moveit/moveit_kinematics/ikfast_kinematics_plugin/scripts/create_ikfast_moveit_plugin.py", line 50, in <module>
    import yaml
ImportError: No module named yaml
```

I imagined it was due to the compatibility with python3 so I tried changing the shebang line of the scritps `create_ikfast_moveit_plugin.py `and `round_collada_numbers.py` to `#!/usr/bin/env python3` and now I get:

```
Traceback (most recent call last):
  File "/home/usr/catkin_ws/src/moveit/moveit_kinematics/ikfast_kinematics_plugin/scripts/create_ikfast_moveit_plugin.py", line 399, in <module>
    main()
  File "/home/usr/catkin_ws/src/moveit/moveit_kinematics/ikfast_kinematics_plugin/scripts/create_ikfast_moveit_plugin.py", line 390, in main
    create_ikfast_package(args)
  File "/home/usr/catkin_ws/src/moveit/moveit_kinematics/ikfast_kinematics_plugin/scripts/create_ikfast_moveit_plugin.py", line 186, in create_ikfast_package
    etree.ElementTree(root).write(f, xml_declaration=True, pretty_print=True, encoding="UTF-8")
  File "src/lxml/etree.pyx", line 2057, in lxml.etree._ElementTree.write
  File "src/lxml/serializer.pxi", line 758, in lxml.etree._tofilelike
  File "src/lxml/etree.pyx", line 318, in lxml.etree._ExceptionContext._raise_if_stored
  File "src/lxml/serializer.pxi", line 682, in lxml.etree._FilelikeWriter.write
TypeError: write() argument must be str, not bytes
```

Which is the correct way to overcome these errors?

### Your environment
* ROS Distro: Noetic
* OS Version: Ubuntu 20.04
* Source build, branch: noetic-devel

### PR DESCRIPTION
**Pull request by rhaschke**
Travis wasn't yet testing `master` branch against Noetic. Switching to Noetic revealed several open issues, which are tackled by this PR:
- Our ikfast unittests were failing and we didn't notice: https://travis-ci.com/github/ros-planning/moveit/jobs/391217949.
- `lxml` API has changed, giving rise to #2322. This PR fixes #2322.
- There were several new warnings, particularly by newer clang compiler.

### CODE REVIEW

File: moveit_kinematics/ikfast_kinematics_plugin/scripts/create_ikfast_moveit_plugin.py [hunk 1]
```diff
@@ -61,7 +61,7 @@
   class InvalidROSPkgException(Exception):
     pass
   def get_pkg_dir(pkg_name):
-    raise InvalidROSPkgException
+    raise InvalidROSPkgException('Failed to localize ROS package {}'.format(pkg_name))
```
Comment (gavanderhoorn) [inline moveit_kinematics/ikfast_kinematics_plugin/scripts/create_ikfast_moveit_plugin.py:64]:
```suggestion
    raise InvalidROSPkgException('Failed to locate ROS package {}'.format(pkg_name))
```

`localize` would be the process of translating the package to another language.

File: moveit_kinematics/ikfast_kinematics_plugin/scripts/create_ikfast_moveit_plugin.py [hunk 2]
```diff
@@ -182,8 +182,7 @@ def create_ikfast_package(args):
     user_name = getuser()
     root.append(xmlElement("maintainer", email="%s@todo.todo" % user_name, text=user_name))
     root.append(xmlElement("buildtool_depend", text="catkin"))
-    with open(pkg_xml_path, 'w') as f:
-      etree.ElementTree(root).write(f, xml_declaration=True, pretty_print=True, encoding="UTF-8")
+    etree.ElementTree(root).write(pkg_xml_path, xml_declaration=True, pretty_print=True, encoding="UTF-8")
```
Comment (gavanderhoorn) [inline moveit_kinematics/ikfast_kinematics_plugin/scripts/create_ikfast_moveit_plugin.py:185]: `etree` now manages opening files itself?

File: moveit_kinematics/test/test_ikfast_plugins.sh [hunk 3]
```diff
@@ -6,6 +6,15 @@
 
 # We will create ikfast plugins for fanuc and panda from moveit_resources
 # using the script auto_create_ikfast_moveit_plugin.sh
+if [ "$ROS_DISTRO" == "noetic" ]; then
+	sudo ln -s /usr/bin/python3 /usr/local/bin/python
```
Comment (gavanderhoorn) [inline moveit_kinematics/test/test_ikfast_plugins.sh:10]: Should we use `update-alternatives` here instead?

**Review by gavanderhoorn — COMMENTED** [review]
Some comments.

File: moveit_kinematics/ikfast_kinematics_plugin/scripts/create_ikfast_moveit_plugin.py (hunk 2 above)
Comment (rhaschke) [reply moveit_kinematics/ikfast_kinematics_plugin/scripts/create_ikfast_moveit_plugin.py:185]: Yes. It supports both file-based writing and opening files itself. However, file-based writing requires to open the file in binary mode: https://bugs.launchpad.net/lxml/+bug/1200352
I thought, just passing the file name is simpler.

File: moveit_kinematics/test/test_ikfast_plugins.sh (hunk 3 above)
Comment (rhaschke) [reply moveit_kinematics/test/test_ikfast_plugins.sh:10]: I tried `alternatives` first, but python wasn't configured for `alternatives`. Not sure, what I missed.

Comment (rhaschke) [reply moveit_kinematics/test/test_ikfast_plugins.sh]: I pushed another commit for this.

**Comment by rhaschke** [comment]
Thanks, Gijs for your comments. This is ready for merging. Please squash these rather independent commits - the scope of this PR quickly grew beyond a mere fix of ikfast tests.

**Comment by felixvd** [comment]
Did you mean to write merge-commit instead of squash?

**Comment by gavanderhoorn** [comment]
I'm indeed assuming @rhaschke meant to write "please *don't* squash".

**Comment by rhaschke** [comment]
Yeah. Of course I mean't **don't** squash... Sorry for the confusion.

**Review by gavanderhoorn — APPROVED** [review]

**Comment by gavanderhoorn** [comment]
No problem.

Congrats on getting your first MoveIt pull request merged and improving open source robotics! :tada:

**Comment by gavanderhoorn** [comment]
@rhaschke: will these changes need backporting to the noetic branch, or will we sync that with `master` before doing a release?

**Comment by rhaschke** [comment]
Thanks for merging. I am keeping master and noetic-devel in sync manually until we really need to diverge...
No need for explicit backporting. But I will backport parts of this to melodic-devel. Stay tuned. :wink:

**Comment by juliamp22** [comment]
Hello!
I have faced the same problem working with noetic, after trying to generate the ik plugin of a .urdf i get the same error:
 File ".../ikfast_kinematics_plugin/scripts/create_ikfast_moveit_plugin.py", line 51, in <module>
    import yaml
ImportError: No module named yaml

Which branch I should be working to ensure that the code works? I tryied master and noetic-devel branch but neither of them worked for me.

**Comment by rhaschke** [comment]
@juliamp22, please file a new issue if you observe a new issue. Do you have installed the Debian package `python3-yaml`?

**Comment by juliamp22** [comment]
Hello @rhaschke yes, I have it installed.
I am trying to use the plugin using the [ikfast_tutorial](https://ros-planning.github.io/moveit_tutorials/doc/ikfast/ikfast_tutorial.html ). 
I have seen that maybe it could be some problems about having python  2 and 3 versions installed at the same time.
Also, which openrave version should I have to use?

Thanks for your help.

**Comment by rhaschke** [comment]
Please open a new issue and exactly describe your environment as well as the step that is failing. Maybe, you use some conda environment, which is typically not compatible to ROS?

**Comment by juliamp22** [comment]
Hello @rhaschke I just opened a new issue, you can find here: https://github.com/ros-planning/moveit/issues/3010
As far as I can see it may be because you are using the OpenRave version openrave0.9 and as far as I can see in ubuntu 20.04 should use the version  0.54.0?

Many thanks for your time and your effort,

Júlia

**Pull request merged**